### ISSUE TITLE
Enhancement: Add labels to fields in JP forms that are valid under WCAG 2.x (a11y)

### ISSUE THREAD

**Issue by mjaragon** [Automattic/jetpack#35356]
### Impacted plugin

Jetpack

### What

JP-forms fields should have a label that passes WCAG 2.x a11y tests

### How

During our A11y tests for a contact form, using different A11y testing tools and the screen reader.
All the different tools we tested, provided the same error related to the `Form label is missing on the message text area.`  (the text area is a Multi-line Text Field)

(see all screenshots & screen recordings in the original issue: https://github.com/a8cteam51/mckenna-academy/issues/97#issuecomment-1819092034 )
 
The suggestion we received from the tools was: "For more information on how to solve the issue please visit this [link](https://dequeuniversity.com/rules/axe/4.8/label?application=AxeChrome)"

When we tested the form with a screen reader, the reader declares all the details about the other fields (Name & Email), but in the `message` section it only reads out the `text area is required`. 

How can we ensure that the contact forms pass all WCAG 2.x a11y tests?

**Comment by jeherve** [comment]
@monsieur-z Do you think you could take a look at this, given your recent work on a11y for forms in #33853?

Thank you!

**Comment by monsieur-z** [comment]
Thanks for reporting this @mjaragon.

I don't have access to the issue you shared. Can you provide more information about how the page was built and how the form is used? For instance, does the page use a Contact Form block? Is a label present in the editor that seems related to the text area?
Also, what combination of screen reader and browser was used?

Spinning up a brand new site and adding the Contact Form block as is to a page, everything seems in order. The text area and label are programmatically associated, and the screen reader (VoiceOver/Chrome) reads the label when the input receives focus.
<img width="500" alt="Screenshot 2024-01-31 at 9 31 11 AM" src="https://github.com/Automattic/jetpack/assets/1620183/d32c00be-3e06-437b-95fb-66a100141a72">
<img width="500" alt="Screenshot 2024-01-31 at 9 31 28 AM" src="https://github.com/Automattic/jetpack/assets/1620183/8c5bd7f5-5267-48d6-bbc5-398beb34628e">

**Comment by jeherve** [comment]
Closing this for now because of the lack of activity on this. We can always reopen in the future if needed.
